### DISCUSSION TITLE
How to set v142 installed by VS2022

### DISCUSSION THREAD

**Discussion by QiuYilin** [microsoft/vcpkg#38616] [Q&A] [answered]
According to the instructions,I  can set the specific version of the compiler in triplet:
```
set(VCPKG_PLATFORM_TOOLSET v142)
set(VCPKG_PLATFORM_TOOLSET_VERSION "14.39.33519")
```

I have installed vc142 by VS2022 build tool,I can find some related folders in  C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519

![image](https://github.com/microsoft/vcpkg/assets/26407388/77c3a0d6-5c1a-4ce4-bdb7-5e127aaf8440)

However when I install packages by vcpkg , it will report

```
error: in triplet x64-windows: Unable to find a valid Visual Studio instance
  with toolset version v142
  with toolset version prefix 14.39.33519
The following Visual Studio instances were considered:
  C:\Program Files\Microsoft Visual Studio\2022\Community
  C:\Program Files\Microsoft Visual Studio\2022\Community
  C:\Program Files\Microsoft Visual Studio\2022\Community
  C:\Program Files\Microsoft Visual Studio\2022\Community
  C:\Program Files\Microsoft Visual Studio\2022\Community
  C:\Program Files\Microsoft Visual Studio\2022\Community
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\
The following paths were examined for Visual Studio instances:
  C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary/Build\vcvarsall.bat
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat
```

Am I missing any steps?

**Comment by StarGate-One** [comment]
> set(VCPKG_PLATFORM_TOOLSET v142)
> set(VCPKG_PLATFORM_TOOLSET_VERSION "14.39.33519")

Try changing VCPKG_PLATFORM_TOOLSET_VERSION to "14.29.30133" and see if that works.

**Comment by QiuYilin** [comment]
@StarGate-One  It can find now , but why?

**Comment by StarGate-One** [comment, marked as answer]
Because the v142 toolset was selected then it was told to use version "14.39.33519" C/C++ version from the v142 toolset, which does not exist.
The first 3 numbers on the version are the toolset. 
You just as easily could have just put in your triplet file `set(VCPKG_PLATFORM_TOOLSET v142)` in which vcpkg would use the latest v142 C/C++ version you have loaded, or `set(VCPKG_PLATFORM_TOOLSET_VERSION "14.29.30133)` and vcpkg would use that version if available otherwise it would error out.

Bottom line you do not need both as you can see a simple mistake can cause vcpkg erratic behavior.

**Reply by QiuYilin** [reply]
Thank you !

**Reply by StarGate-One** [reply]
You are most welcome! Glad to have helped.
